client/client-multi/client-c/src/kaa/kaa_logging.c

```diff
@@ -420,56 +479,79 @@ static void place_record_in_next_bucket(kaa_log_collector_t *self, kaa_log_recor
     new_record->bucket_id = self->last_bucket.id + 1;
 }
 
-// Creates new bucket
-static void form_new_bucket(kaa_log_collector_t *self)
+// Puts record in the current bucket
+static void place_record_in_current_bucket(kaa_log_collector_t *self, kaa_log_record_t *new_record)
+{
+    new_record->bucket_id = self->last_bucket.id;
+    self->last_bucket.log_count++;
+}
+
+// Creates a new bucket and places an initial record there if given
+static void form_new_bucket(kaa_log_collector_t *self, const kaa_log_record_t *initial_record)
 {
     self->last_bucket.id++;
     // Avoid hitting reserved values
     if (!self->last_bucket.id) {
         self->last_bucket.id++;
     }
 
-    self->last_bucket.log_count = 0;
-    self->last_bucket.size = 0;
+    if (initial_record) {
+        self->last_bucket.log_count = 1;
+        self->last_bucket.size = initial_record->size;
```

Comment:
Could you document usage of `last_bucket_info::size`. It's not clear to my why size is set here but not increased at `place_record_in_current_bucket()`.

I would expect that
```c
form_new_bucket(self, NULL);
place_record_in_current_bucket(self, record);
```
is equal to
```c
form_new_bucket(self, record);
```